app/helpers/declarable_helper.rb

```diff
@@ -40,23 +40,28 @@ def declarable_stw_link(declarable, search)
   end
 
   def declarable_back_link
-    link_to('Back', declarable_path, class: 'govuk-back-link')
+    link_to('Back', goods_nomenclature_path, class: 'govuk-back-link')
   end
 
   def declarable_link
-    link_to("Return to #{current_declarable_code}", declarable_path, class: 'govuk-link')
+    link_to("Return to #{current_goods_nomenclature_code}", goods_nomenclature_path, class: 'govuk-link')
   end
 
-  def declarable_path
-    if current_declarable_code.size == Heading::SHORT_CODE_LENGTH
-      heading_path(declarable_path_opts)
+  def goods_nomenclature_path
+    case current_goods_nomenclature_code&.size
+    when nil
+      home_path
+    when Chapter::SHORT_CODE_LENGTH
+      chapter_path(goods_nomenclature_path_opts)
+    when Heading::SHORT_CODE_LENGTH
+      heading_path(goods_nomenclature_path_opts)
     else
-      commodity_path(declarable_path_opts)
+      commodity_path(goods_nomenclature_path_opts)
```

Comment:
it looks the path can be a method inside
Chapter, Heading and commodity
and we'd not need the case